zip-0226.rst

```diff
@@ -1,8 +1,313 @@
+- The Merkle Path Validity should check the existence of the note commitment as usual (and not like with dummy notes):
+    - Check that (path, pos) is a valid Merkle path of depth :math:`\mathsf{MerkleDepth^Orchard}`, from :math:`\mathsf{cm^old}` to the anchor :math:`\mathsf{rt^Orchard}`.
+- The Nullifier Integrity will be changed to prevent the identification of notes
+    - Replace the :math:`\psi_{old}` value with a generic :math:`\psi'` as :math:`\mathsf{nf_old = DeriveNullifier_nk}(\rho^\mathsf{old}, \psi', \mathsf{cm^old})`
+    - if :math:`split = 0` then constrain :math:`\psi' = \psi^{old}`. (Otherwise :math:`\psi'` should be sampled randomly.) 
+
+**Enabling Backwards Compatibility with ZEC Notes:** the following constraints must be added to enable backwards compatibility with the Orchard ZEC notes.
+
+The old note commitment is computed using a “rolling-aggregate” Sinsemilla commitment. This means that the commitment is computed by adding new chunks or windows to the accumulated value. This method will be used in order to maintain a single commitment instance for the old note commitment, that will be used both for Orchard ZEC notes and for ZSA notes. The original Orchard ZEC notes will be conserved and not actually be converted into ZSA notes, as we will always need to compute them.
+
+- The input note in the old note commitment integrity must either include an Asset identifier (ZSA note) or not (ZEC-Orchard note)
+    - If the Asset identifier auxiliary input is set :math:`\mathsf{AssetId}` = :math:`\mathcal{V}^\mathsf{Orchard}`
+        - NoteCommitment has a “compatibility” path that computes the note commitment as in plain Orchard (i.e.: without including the identifier)
+        - This path also uses the original domain separator for ZEC note commitment
+    - Else, 
+        - The NoteCommitment adds the identfier, :math:`\mathsf{AssetId}`, as a final “chunk” of the Sinsemilla commitment
+        - The NoteCommitment uses a different domain separator for ZSA note commitment
+
+
+Backward Compatibility
+----------------------
+
+In order to have a "clean" backwards compatibility with the ZEC notes, we have designed the circuit to support both ZEC and ZSA notes. As we specify above, there are three main reasons we can do this:
+- The input notes with an Asset identifer denote the custom Assets, generating a note commitment that includes the Asset identifer; whereas the notes without an identifier, denote the ZEC notes, and generate a note commitment that does not include the Asset identifier, in order to maintain the referencability to the Merkle tree
+- The value commitment is abstracted to allow for the value base-point as a variable private input to the proof
+- The ZEC-based actions will still include dummy input notes, whereas the ZSA-based actions will include split input notes.
+
+
+Burn Mechanism
```

Comment:
Can be restructured to have the specification and then its rationale immediately after spec, in one section

Reply:
Done